vertexai/function-calling/functioncalling.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// function-calling shows an example of OpenAPI Function Call
+//
+// This Function Call involves 3 messages:
+// - ask the model to generate a Function Call request
+// - call the Open API service (simulated in this example)
+// - ask the model to interpret the Function Call response
+package main
+
+// [START aiplatform_gemini_function_calling]
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+func main() {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	location := "us-central1"
+	modelName := "gemini-pro"
+
+	prompt := "What's the weather like in Paris?"
+
+	if projectID == "" {
+		log.Fatal("require environment variable GOOGLE_CLOUD_PROJECT")
+	}
+
+	err := weather(os.Stdout, prompt, projectID, location, modelName)
```

Comment:
Pass in a writer instead of stdout. We can't read from stdout in tests (unless we do some awkward pipe-ing).